clients/admin-ui/src/features/common/hooks/useOutsideClick.ts

```diff
@@ -0,0 +1,22 @@
+import { LegacyRef, useEffect, useRef } from "react";
+
+// eslint-disable-next-line import/prefer-default-export
+export const useOutsideClick = (handleClick: () => void) => {
+  const ref = useRef<HTMLDivElement | undefined>(undefined) as
+    | LegacyRef<HTMLDivElement>
```

Comment:
This is casted to this type because the Chakra UI `Box` component expects to be `LegacyRef<HTMLDivElement> | undefined`